examples/bzlmod/MODULE.bazel

```diff
@@ -1,5 +1,7 @@
 module(name = "example")
 
+bazel_dep(name = "rules_cc", version = "0.0.9")
```

Comment:
It looks like we don't need this.

Reply:
Unlike the workspace example, in the bzlmod example, we do load cc_binary from rules_cc, so we need to declare it here.

We can remove that load in favor of native cc_binary, but I think it's better this way. WDYT?

Reply:
This is fine, although we should probably load a more recent version (that's still below `0.1.0`) to ensure compatibility with Bazel 8 and future versions.